File: lib/src/main/java/com/sd/lib/cache/handler/impl/BooleanHandler.java
package com.sd.lib.cache.handler.impl;

import com.sd.lib.cache.CacheInfo;
import com.sd.lib.cache.handler.BaseCacheHandler;

/**
 * Boolean处理类
 */
public class BooleanHandler extends BaseCacheHandler<Boolean> {
    public BooleanHandler(CacheInfo info) {
        super(info);
    }

    @Override
    protected byte[] valueToByte(Boolean value) {
        return value.toString().getBytes();
    }

    @Override
    protected Boolean byteToValue(byte[] bytes, Class<?> clazz) {
        return Boolean.valueOf(new String(bytes));
    }

    @Override
    protected String getKeyPrefix() {
        return "boolean_";
    }
}

File: lib/src/main/java/com/sd/lib/cache/handler/impl/BooleanHandler.kt
package com.sd.lib.cache.handler.impl

import com.sd.lib.cache.CacheInfo
import com.sd.lib.cache.handler.BaseCacheHandler

/**
 * Boolean处理类
 */
internal class BooleanHandler(info: CacheInfo) : BaseCacheHandler<Boolean>(info, "boolean") {
    override fun encodeToByteImpl(value: Boolean, clazz: Class<*>?): ByteArray {
        return value.toString().toByteArray()
    }

    override fun decodeFromByteImpl(bytes: ByteArray, clazz: Class<*>?): Boolean {
        return String(bytes).toBooleanStrict()
    }
}